dao/priorityrequest/priority_request.go

```diff
@@ -119,16 +122,19 @@ func (m *defaultPriorityRequestModel) GetLatestHandledRequestId() (requestId int
 }
 
 func (m *defaultPriorityRequestModel) UpdateHandledPriorityRequestsInTransact(tx *gorm.DB, requests []*PriorityRequest) (err error) {
-	ids := make([]uint, 0, len(requests))
 	for _, request := range requests {
-		ids = append(ids, request.ID)
-	}
-	dbTx := tx.Table(m.table).Where("id in ?", ids).Update("status", HandledStatus)
-	if dbTx.Error != nil {
-		return dbTx.Error
-	}
-	if dbTx.RowsAffected != int64(len(ids)) {
-		return types.DbErrFailToUpdatePriorityRequest
+		dbTx := tx.Table(m.table).Where("id = ?", request.ID).Updates(
+			map[string]interface{}{
+				"status":     HandledStatus,
+				"l2_tx_hash": request.L2TxHash,
+			},
+		)
+		if dbTx.Error != nil {
+			return dbTx.Error
+		}
+		if dbTx.RowsAffected != 1 {
```

Comment:
Why `dbTx.RowsAffected` must equal to 1 when update batch priority requests?

Reply:
Because the implementation changed from batch to one-by-one.
To update the txHash to the correspond request, need use loop to update each request one by one.
Because it updates one by one, the `dbTx.RowsAffected` should be 1.

Reply:
Because each record has different l2_tx_hash, can't do batch updates here